### PR TITLE
feat(work): gate re-entry on completed state with idempotent short-circuits (GH-398)

### PR DESCRIPTION
**Pull request by thomfilg**
## What this PR fixes (scope: ECHO-4552 Issue 2 only)

The `/work` dispatcher could not advance a ticket whose workflow was already in the terminal completed state — re-running `work-next.js` against a state file with `status: "completed"` and `stepStatus.complete: "completed"` fell through to the normal planning loop instead of returning `action: "complete"`.

This PR adds a single dispatcher-level early-return for that exact state, plus an audit-trail fingerprint on gate transitions, plus a PR-merged short-circuit so a merged remote PR can advance a stuck local state.

**Out of scope for this PR:**
- ECHO-4552 Issue 1 (FAILED-gate not re-parsing after an in-place fix). Symptom is the opposite direction (a failed gate caching its verdict), tracked separately — see follow-up issue.
- Per-gate idempotent short-circuits. An earlier draft of this PR added them; they were reverted because they risked skipping re-parse in in-progress workflows where the user may have edited inputs between gate satisfaction and resume.
- Content-hash drift detection on gate inputs. Tracked separately — see follow-up issue.

## Changes

- **`scripts/workflows/work/work-next.js`** — single dispatcher early-return when **both** `state.status === "completed"` **and** `state.stepStatus.complete === "completed"`. Releases the session guard and returns the terminal `action: "complete"` payload.
- **`scripts/workflows/work/work-next.js`** — fail-open PR-merged probe: `gh pr view --json state` against the ticket worktree; on `{ state: "MERGED" }`, marks the local state as completed and returns the terminal payload. Any `gh` error (non-zero exit, missing binary, auth failure, malformed JSON, missing worktree dir) is swallowed; the workflow falls through to existing behavior.
- **`scripts/workflows/work/engine/transition-step.js`** — when a `*_gate` step transitions to `completed`, writes `ws.gateFingerprints[<gate>] = { pluginVersion, satisfiedAt }` as a non-load-bearing audit trail. Existing state files without `gateFingerprints` continue to load unchanged.

## What this PR does NOT change

- `spec_gate` and `tasks_gate` handlers continue to re-parse their inputs on every invocation. Parsing is fast and the only safe behavior in the presence of in-progress edits.
- The `complete` step semantics, the workflow state machine, and `transition-step.js`'s gate ordering are untouched.

## Testing Plan

```
node --test scripts/workflows/work/__tests__/work-next-resume.test.js
node --test scripts/workflows/work/__tests__/work-next-pr-merged.test.js
node --test scripts/workflows/work/engine/__tests__/transition-step-gate-fingerprint.test.js
```

Coverage:
- Dispatcher short-circuit fires when both `status: completed` AND `stepStatus.complete: completed`.
- Dispatcher short-circuit does **not** fire when `stepStatus.complete` is `pending` (partial-completion state).
- Two consecutive invocations on unchanged completed state produce byte-identical output (modulo `lastPlanTimestamp`).
- PR-merged probe advances to `complete` on `{ state: "MERGED" }`; falls through on `gh` failure; skips the probe entirely when the worktree dir is missing.
- `gateFingerprints` is written for `*_gate → completed` transitions and preserves prior entries; non-gate transitions write nothing.

## Dev Checks
- [Y] Functionality can be toggled on/off (dispatcher branch is conditional on persisted state)
- [Y] New code is covered by unit/integration tests
- [ ] Breaking changes for the API have been inserted into a deprecation cycle (N/A)
- [ ] There is appropriate documentation for the new work (N/A — internal orchestrator change)
- [ ] Any new environment variables are populated into variable groups for all environments (N/A — no new env vars)

---

> [!NOTE]
> **Medium Risk**
> Touches core `/work` orchestration flow by adding new early-return paths and state mutation based on `gh pr view`, which could incorrectly mark workflows completed if edge cases slip through (though guarded and fail-open). Changes are well-covered by new tests, reducing but not eliminating risk.
> 
> **Overview**
> Fixes `/work` resume behavior by adding a dispatcher-level terminal short-circuit in `work-next.js`: if the persisted state has `status: "completed"`, it immediately returns `action: "complete"` and releases the session guard (even for older state-file shapes where `stepStatus.complete` is missing/pending).
> 
> Adds a fail-open **PR merged** probe in `work-next.js` that runs `gh pr view --json state` in the ticket worktree; when it reports `MERGED`, the local workflow state is force-marked completed (all `stepStatus` set to `completed`) and the terminal payload is returned, while skipping the probe entirely if the worktree dir is missing.
> 
> Enhances `transition-step.js` to record an audit-only `gateFingerprints[<*_gate>] = { pluginVersion, satisfiedAt }` whenever a gate step transitions to `completed`, and changes `work-next.js` stdout to emit single-line JSON for easier parsing. New node tests cover the short-circuits, fail-open behavior, worktree-missing safety, and gate fingerprint persistence.
> 
> <sup>Reviewed by [Cursor Bugbot](https://cursor.com/bugbot) for commit d92f2ef799d958e6ba713ced43e37acbf9bc0920. Bugbot is set up for automated code reviews on this repo. Configure [here](https://www.cursor.com/dashboard/bugbot).</sup>

### CODE REVIEW

File: scripts/workflows/work/work-next.js [hunk 1]
```diff
@@ -472,11 +472,17 @@ function getNextInstruction(ticketRaw, rework) {
...
+  //   - state.status === 'completed' (overall workflow flag), AND
+  //   - state.stepStatus.complete === 'completed' (terminal step flag)
+  // Both conditions together — never one without the other. Anything weaker
+  // risks short-circuiting in-progress workflows where the user may have
+  // edited inputs between gate satisfaction and resume.
   const _preCheckState = loadWorkState(safeName);
   if (
     _preCheckState &&
-    getCurrentStep(_preCheckState) === 'complete' &&
+    _preCheckState.status === 'completed' &&
```
[the remark this answers is not in the document]
Comment (thomfilg) [reply scripts/workflows/work/work-next.js]: Fixed in d92f2ef7: loosened to `state.status === 'completed'` alone per spec P0 #1, matching the older state-file shape from ECHO-4552 Issue 2. Added regression tests for both `stepStatus.complete: pending` and absent-key shapes.
